### ISSUE TITLE
Why is in-browser link navigation a non goal?

### ISSUE THREAD

**Issue by mgiuca** [WICG/pwa-url-handler#9]
Under Non-Goals:

> Launching PWAs due to in-browser link navigation. Instead, this explainer focuses on scenarios where the URL activation has occurred from outside of the browser.

Why make this distinction? It seems to unnecessarily distinguish native and web apps as sources. (For example, clicking a link to a native app would trigger the URL handler; clicking a link from a PWA would not.)

Your use case says:

> ... in their native e-mail application, (eg., https://open.spotify.com/album/7FA9xfqPrBaja1sEv15DU2 in the Outlook app) ...

But if the user was using the outlook.com PWA, this would not work? Why not?

**Comment by LuHuangMSFT** [comment]
We initially wanted the focus of this explainer to be on the benefits of having a declarative URL handling API and association scheme. There is no strong reason not to also account for in-browser and in-app-window link navigation but we would have had to detail some intricacies that were not the focus of this explainer. 

Eg. If a user deliberately navigates to an in-scope URL of an installed PWA using the address bar, then clicks on a link that is also in-scope, should an app window be opened? 

If we remove in-browser link navigation as a non-goal, how much more detail would you like to see about that? Would it be sufficient to leave most of that behavior up to the UA instead of specifying it?

**Comment by mgiuca** [comment]
> There is no strong reason not to also account for in-browser and in-app-window link navigation but we would have had to detail some intricacies that were not the focus of this explainer.

Right, but those intricacies (mostly, figuring out whether or not to do link capturing in any of the ~20 different ways to navigate) are not specific to this proposal. I don't think they need to be captured by this proposal (they are common to all of the link capturing discussions, e.g., sw-launch).

This is something that we (Chrome PWA team) have spent an extensive amount of time researching and designing (I'll try and find a link later). I'm not sure if it needs to be part of a spec, or just leave it up to the UA behaviour, but suffice to say it's quite complex and I think it's fine to leave it out of this proposal. I think this proposal should focus just on defining the _set of URLs to be link-captured_, and leave the actual behavioural requirements around link capturing to the existing efforts.
